### PR TITLE
Publish amd64 and arm64 awx image on release

### LINKED ISSUE #14643: ARM64 images built and Available on AWX repository
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

New Feature

### Feature Summary

The feature I'm requesting is officially built ARM images. I note that support has been added to build the images yourself, however not an option for a lot of people out there.

Implementation of this proposal requires no additional infrastructure, just some minor changes to the build commands.

Currently docker has buildx. This tool enables cross-platform building of containers for countless architectures. best of all, this occurs within a single build command, `docker buildx build --platform=`. Using this command instead of the original `docker build`. cross-compiles for each of the platforms listed and places all images within the same manifest. In addition, you can prepend `--push` to push all built images and the manifest to the container registry.

I currently use this method to cross-compile all of my container images.

What requires changing

docker build command. 
``` diff
-docker build -t {{ awx_image }}:{{ awx_image_tag }} \
+docker buildx build -t {{ awx_image }}:{{ awx_image_tag }} \
        -f {{ dockerfile_name }} \
        --build-arg VERSION={{ awx_version }} \
        --build-arg SETUPTOOLS_SCM_PRETEND_VERSION={{ awx_version }} \
        --build-arg HEADLESS={{ headless }} \
+      --platforms=linux/amd64,linux/arm64
+      --push
        .
```
notes:
- _` --platforms=linux/amd64,linux/arm64` this will build both amd64 and arm64 together and place them in a single manifest_
- _`--push` this will push everything together to the container registry_

Every `FROM` declaration within dockerfiles
``` diff
+ARG TARGETPLATFORM=linux/amd64,linux/arm64

-FROM quay.io/centos/centos:stream9 as builder
+FROM --platform=$TARGETPLATFORM quay.io/centos/centos:stream9 as builder

```
notes:
- _`ARG TARGETPLATFORM=linux/amd64,linux/arm64` sets ` TARGETPLATFORM` variable to have a default value if not specified at runtime_
- _`FROM --platform=$TARGETPLATFORM` tells docker to use the specified architecture for the container. if this value is omitted from the `FROM` declaration, the build systems architecture is used._

For cross compilation to work packages binfmt-support and qemu-user-static (these are the debian package name). Both together allow the running of binaries of a different architecture. Or you can do the build from a docker container (the method I use), which contains everything required fro the cross-compilation to work. Prior to building you have to activate modules in the kernal for other binaries to run. `update-binfmts --enable`

I'm not familiar of how the github CI/CD pipelines work, however I am successfully doing cross-compilation within the gitlab ecosystem. their runners are all AMD64. You maybe able to convert this stripped down gitlab ci job. Original [here](https://gitlab.com/nofusscomputing/projects/gitlab-ci/-/blob/development/docker/build.gitlab-ci.yaml)

``` yaml
.build_docker_container:
  stage: build
  image: 
    name: nofusscomputing/docker-buildx-qemu:dev
    pull_policy: always
  services:
    - name: docker:23-dind
      entrypoint: ["env", "-u", "DOCKER_HOST"]
      command: ["dockerd-entrypoint.sh"]
  variables:
    DOCKER_HOST: tcp://docker:2375/
    DOCKER_DRIVER: overlay2
    DOCKER_DOCKERFILE: dockerfile
    # See https://github.com/docker-library/docker/pull/166
    DOCKER_TLS_CERTDIR: ""
  before_script:
    - git submodule foreach git submodule update --init
    - if [ "0$JOB_ROOT_DIR" == "0" ]; then ROOT_DIR=gitlab-ci; else ROOT_DIR=$JOB_ROOT_DIR ; fi
    - echo "[DEBUG] ROOT_DIR[$ROOT_DIR]"
    - docker info
    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
    - pip3 install setuptools wheel
      # see: https://gitlab.com/gitlab-org/gitlab-runner/-/merge_requests/1861 
      # on why this `docker run` is required. without it multiarch support doesnt work.
    - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
    - update-binfmts --display
    - update-binfmts --enable # Important: Ensures execution of other binary formats is enabled in the kernel
    - docker buildx create --driver=docker-container --driver-opt image=moby/buildkit:v0.11.6 --use
    - docker buildx inspect --bootstrap

  script: 
    - update-binfmts --display
    - |
        docker buildx build --platform=$DOCKER_IMAGE_BUILD_TARGET_PLATFORMS . \
          --label org.opencontainers.image.created="$(date '+%Y-%m-%d %H:%M:%S%:z')" \
          --label org.opencontainers.image.documentation="$CI_PROJECT_URL" \
          --label org.opencontainers.image.source="$CI_PROJECT_URL" \
          --label org.opencontainers.image.revision="$CI_COMMIT_SHA" \
        --push \
        --build-arg CI_JOB_TOKEN=$CI_JOB_TOKEN --build-arg CI_PROJECT_ID=$CI_PROJECT_ID --build-arg CI_API_V4_URL=$CI_API_V4_URL \
        --file $DOCKER_DOCKERFILE \
        --tag $DOCKER_IMAGE_BUILD_REGISTRY/$DOCKER_IMAGE_BUILD_NAME:$DOCKER_IMAGE_BUILD_TAG;

        docker buildx imagetools inspect $DOCKER_IMAGE_BUILD_REGISTRY/$DOCKER_IMAGE_BUILD_NAME:$DOCKER_IMAGE_BUILD_TAG;

```
summary:
- starts a docker container `nofusscomputing/docker-buildx-qemu:dev` where all commands are run from, including build
- as the container is dind, links to docker via socket
- for containers that use python packages, `setuptools` and `wheel` wheel are required if the package requires compilation
- `update-binfmts --enable` enable kernal modules
- `docker buildx create --driver=docker-container --driver-opt image=moby/buildkit:v0.11.6 --use` sets up buildx to use buildkit
- `docker buildx build {etc}...` all in one command to cross-compile build, manifest creation and push to a container registry .
- the final `inspect` shows the manifest and containing images.

I'm happy to assist or if ok, start a PR. However the latter I will require someone with Github action knowledge to walk me through adjusting.

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Steps to reproduce

.

### Current results

.

### Sugested feature result

That https://quay.io/repository/ansible/awx contains both amd64 and arm64 images. Yes, I know different repos, same same for the operator and ansible-ee images.

### Additional information

_No response_

### LINKED ISSUE #15079: ARM64 system - AWX image pull failing
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

I might need a clarification maybe, more than technical help as I am probably doing something wrong.
My deployment of AWX via AWX-Operator is failing on an ARM64 CPU system, despite pulling an AWX image that is supposed to be compatible with AMD64 (quay.io/ansible/awx:23.9.0)

The awx-web pod (kubernetes) fails to find a suitable image.
But since AWX 23.8.1 (https://github.com/ansible/awx/releases) or 23.9.0 (https://forum.ansible.com/t/announcing-awx-23-9-0-and-awx-operator-2-12-2/4178) ARM64 system should have an image... correct ?

```
Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  21m                 default-scheduler  Successfully assigned awx/awx-web-55d5d78c97-hfdw4 to ip-xxx
  Normal   Pulled     21m                 kubelet            Container image "quay.io/ansible/awx-ee:latest" already present on machine
  Normal   Created    21m                 kubelet            Created container init
  Normal   Started    21m                 kubelet            Started container init
  Normal   Pulled     21m                 kubelet            Container image "quay.io/centos/centos:stream9" already present on machine
  Normal   Created    21m                 kubelet            Created container init-projects
  Normal   Started    21m                 kubelet            Started container init-projects
  Normal   Pulled     21m                 kubelet            Container image "docker.io/redis:7" already present on machine
  Normal   Created    21m                 kubelet            Created container redis
  Normal   Started    21m                 kubelet            Started container redis
  Normal   Pulling    20m (x2 over 21m)   kubelet            Pulling image "quay.io/ansible/awx:23.9.0"
  Warning  Failed     20m (x2 over 21m)   kubelet            Failed to pull image "quay.io/ansible/awx:23.9.0": rpc error: code = NotFound desc = failed to pull and unpack image "quay.io/ansible/awx:23.9.0": no match for platform in manifest: not found
  Warning  Failed     20m (x2 over 21m)   kubelet            Error: ErrImagePull
  Normal   BackOff    20m (x3 over 21m)   kubelet            Back-off pulling image "quay.io/ansible/awx:23.9.0"
  Warning  Failed     20m (x3 over 21m)   kubelet            Error: ImagePullBackOff
  Warning  Failed     20m (x2 over 21m)   kubelet            Error: ImagePullBackOff
  Normal   BackOff    61s (x85 over 21m)  kubelet            Back-off pulling image "quay.io/ansible/awx:23.9.0"
```
  
  This is the CPU on this server
```
  $ lscpu
Architecture:             aarch64
  CPU op-mode(s):         32-bit, 64-bit
  Byte Order:             Little Endian
CPU(s):                   2
  On-line CPU(s) list:    0,1
Vendor ID:                ARM
  Model name:             Neoverse-N1
    Model:                1
    Thread(s) per core:   1
    Core(s) per socket:   2
    Socket(s):            1
    Stepping:             r3p1
    BogoMIPS:             243.75
```

Am I missing an important detail here, or, I'm too impatient and this work needs to be completed first https://github.com/ansible/awx/issues/14643 ?
Thank you all.

### AWX version

23.9.0

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

kubernetes

### Modifications

no

### Ansible version

NA

### Operating system

Amazon Linux 2023

### Web browser

_No response_

### Steps to reproduce

Run AWX operator install similar to how it was successfully installing AWX on ADM64 systems.

### Expected results

A successful install of AWX components.

### Actual results

awx web pod cannot be completed.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
##### SUMMARY
- change CI to use `make awx-kube-build` instead of build playbook
- update staging CI to build and push multiarch awx image
- update doc to use `make awx-kube-build` to build awx image
- remove build playbook (no longer used)

Resolves https://github.com/ansible/awx/issues/14643
Resolves https://github.com/ansible/awx/issues/15079
##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Other

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -94,11 +94,10 @@ jobs:
       - name: Build AWX image
         working-directory: awx
         run: |
-          ansible-playbook -v tools/ansible/build.yml \
-            -e headless=yes \
-            -e awx_image=awx \
-            -e awx_image_tag=ci \
-            -e ansible_python_interpreter=$(which python3)
+          VERSION=`make version-for-buildyml` make awx-kube-build
```
Comment (rooftopcellist) [inline .github/workflows/ci.yml:97]: Should this be `version-for-build-yml`

**Review by rooftopcellist — APPROVED** [review]

**Comment by kurokobo** [comment]
Hi, thanks for working on this!
Are there any actual logs for this GHA and available (testable) images built by it?

**Review by shanemcd — APPROVED** [review]

**Comment by TheRealHaoLiu** [comment]
I'm working on getting this working on my fork so i can test it...

**Comment by TheRealHaoLiu** [comment]
if anyone want to watch the paint dry here's where im rehearsing the staging process...
https://github.com/TheRealHaoLiu/awx/actions/workflows/stage.yml

**Comment by TheRealHaoLiu** [comment]
```
#38 [auth] therealhaoliu/awx-operator:pull,push token for ghcr.io
#38 DONE 0.0s
#37 exporting to image
#37 pushing layers 0.3s done
#37 ERROR: failed to push ghcr.io/therealhaoliu/awx-operator:100.1.2: unexpected status from POST request to https://ghcr.io/v2/therealhaoliu/awx-operator/blobs/uploads/: 403 Forbidden
------
 > exporting to image:
------
ERROR: failed to solve: failed to push ghcr.io/therealhaoliu/awx-operator:100.1.2: unexpected status from POST request to https://ghcr.io/v2/therealhaoliu/awx-operator/blobs/uploads/: 403 Forbidden
make: [Makefile:121: docker-buildx] Error 1 (ignored)
```

latest failure...

on my fork im not able to push to `ghcr.io/therealhaoliu/awx-operator` for some reason

**Comment by TheRealHaoLiu** [comment]
in the same workflow im able to push ghcr.io/therealhaoliu/awx and ghcr.io/therealhaoliu/awx-ee

**Comment by kurokobo** [comment]
@TheRealHaoLiu 
> on my fork im not able to push to ghcr.io/therealhaoliu/awx-operator for some reason

Thanks, maybe you have to assign `Write` perm to `therealhaoliu/awx` on "Package settings" on `therealhaoliu/awx-operator` (maybe here: https://github.com/users/therealhaoliu/packages/container/awx-operator/settings)?

**Comment by TheRealHaoLiu** [comment]
ur absolutely right @kurokobo 🥂

**Comment by TheRealHaoLiu** [comment]
14th time is the charm... the latest staging worked https://github.com/TheRealHaoLiu/awx/actions/runs/8560172659/job/23458535766 
NOTE: create draft release of AWX operator failed bc i didnt give `AWX_OPERATOR_RELEASE_TOKEN` it should be fine on AWX repo

NOTE: stage workflow does take significantly longer than before due to cross platform build  increase from 30 min to 1 hr

here's the images that the stage workflow produced 
- ghcr.io/therealhaoliu/awx:100.100.101
- ghcr.io/therealhaoliu/awx-operator:100.100.101
- ghcr.io/therealhaoliu/awx-ee:100.100.101

tested locally on microshift it works

**Comment by kurokobo** [comment]
@zaki-lknr
If you are interested in this, you can test above images before oficial releases😃

**Comment by TheRealHaoLiu** [comment]
```
apiVersion: awx.ansible.com/v1beta1
kind: AWX
  name: awx
  namespace: awx
spec:
  control_plane_ee_image: ghcr.io/therealhaoliu/awx-ee:100.100.101
  ee_images:
  - image: quay.io/ansible/awx-ee:latest
    name: AWX EE (latest)
  - image: ghcr.io/therealhaoliu/awx-ee:100.100.101
    name: AWX EE (100.100.101)
  image: ghcr.io/therealhaoliu/awx
  image_version: 100.100.101
  init_container_image: ghcr.io/therealhaoliu/awx-ee
  init_container_image_version: 100.100.101
```

**Pull request merged**